### PR TITLE
Fixed content length handling

### PR DESCRIPTION
**Pull request by jcellary**
Fixed validation of content size - max image size is 10 mb

When information about file size is not provided in a header, I read the whole stream, to check it's size. As ugly as it may seem at first, I'm actually using a buffed input stream, to buffer the read content and not to process it multiple times. Also I need to read the whole image to memory anyway, as I'm processing it twice - first for validation purposes and secondly to save it to S3

pinging @sqreek

### CODE REVIEW

File: service/user-avatar/src/main/java/com/wikia/services/useravatar/resources/UserAvatarResource.java [hunk 1]
```diff
@@ -296,11 +323,12 @@ public UserAvatarCreatedResult createUserAvatar(
   })
   public UserAvatarCreatedResult updateOrCreateUserAvatar(
       @Auth AuthenticatedUser user,
+      @HeaderParam("user-agent") String userAgent,
       @PathParam("userId") int userId,
       @FormDataParam("file") InputStream fileInputStream,
       @FormDataParam("file") FormDataContentDisposition fileDetail) {
     ensureAuthUserIdIsSameAsUserId(userId, user);
-    return doUpdateOrCreateUserAvatar(user.getUserId(), userId, fileInputStream, fileDetail);
+    return doUpdateOrCreateUserAvatar(user.getUserId(), userId, fileInputStream, fileDetail, userAgent);
```
Comment (sqreek) [inline service/user-avatar/src/main/java/com/wikia/services/useravatar/resources/UserAvatarResource.java]: updateOrCreateUserAvatar

**Comment by sqreek** [comment]
getAndValidateFileSize and validateSize is duplicated code.
Move it to library, create validationService, move responsibility to one of services

**Comment by jcellary** [comment]
@sqreek please take a second look

File: lib/service-core/src/main/java/com/wikia/pandora/core/util/StreamUtils.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.wikia.pandora.core.util;
+
+import net.logstash.logback.encoder.org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+
+public class StreamUtils {
```
Comment (WikiaTech) [inline lib/service-core/src/main/java/com/wikia/pandora/core/util/StreamUtils.java]: MAJOR Add a private constructor to hide the implicit public one. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1118)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by sqreek** [comment]
fix sonar comment and LGTM

**Pull request merged**